### ISSUE TITLE
Split up code for improved performance.

### ISSUE THREAD

**Issue by dannyvankooten** [baselbers/woocommerce-pdf-invoices#60]
Right now, the plugin loads classes like `BEWPI_General_Settings` which is only required for requests to `/wp-admin/`. 

It does several other things like calling `update_option` on `init` which would cause a SQL write query for every page load. Not a huge issue as WordPress core checks if the option value actually changed and otherwise does nothing but it's still unnecessary code.

It would be neat if code could be split-up in a way so that for public requests, only the minimum amount of code runs. For sites with a ton of visitors, this makes a huge difference.

**Comment by dannyvankooten** [comment]
PS. I could be wrong about this but it seems the entire plugin doesn't need to be loaded for public facing requests?

**Comment by baselbers** [comment]
You are right! This will be top prio. The plugin does need to load for public requests for example to download invoice from "My Account" page.
